Source/Core/VideoCommon/PixelShaderGen.cpp

```diff
@@ -319,6 +320,21 @@ PixelShaderUid GetPixelShaderUid()
   uid_data->fog_proj = bpmem.fog.c_proj_fsel.proj;
   uid_data->fog_RangeBaseEnabled = bpmem.fogRange.Base.Enabled;
 
+  BlendingState state = {};
```

Comment:
Probably won't make a huge difference, but you could wrap all this in a if (bpmem.dstalpha.enable), avoiding the call to Generate. Not sure it's worth the two sources of "dual src" though, and we'll likely clean this up with "pipeline UIDs" in the future anyway.